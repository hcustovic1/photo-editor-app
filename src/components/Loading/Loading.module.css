.loaderContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh;
    color: var(--color-text);
    background-color: var(--color-background);
  }
  
  .spinner {
    border: 4px solid var(--color-background);
    border-top: 4px solid var(--color-primary);
    border-radius: 50%;
    width: 40px;
    height: 40px;
    animation: spin 1s linear infinite;
    margin-bottom: 16px;
  }
  
  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
  
  p {
    font-size: 1.2rem;
    font-weight: var(--font-weight-bold);
    margin: 0;
  }