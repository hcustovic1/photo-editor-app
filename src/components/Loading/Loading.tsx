import React from 'react';
import styles from './Loading.module.css';

export const Loading: React.FC = () => {
  return (
    <div className={styles.loaderContainer} role="alert" aria-busy="true">
      <div className={styles.spinner} aria-hidden="true"></div>
      <p>Loading...</p>
    </div>
  );
};
