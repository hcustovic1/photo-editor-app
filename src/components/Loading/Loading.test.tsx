import { render } from '@testing-library/react';
import { describe, test, expect } from 'vitest';
import { Loading } from './Loading';

describe('Loading Component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<Loading />);

    expect(asFragment()).toMatchSnapshot();
  });
});
