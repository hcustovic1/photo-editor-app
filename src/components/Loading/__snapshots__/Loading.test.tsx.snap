// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    class="_loaderContainer_55dd7a"
    role="alert"
  >
    <div
      aria-hidden="true"
      class="_spinner_55dd7a"
    />
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;

exports[`Loading component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    class="_loaderContainer_55dd7a"
    role="alert"
  >
    <div
      aria-hidden="true"
      class="_spinner_55dd7a"
    />
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
